### DISCUSSION TITLE
i have a problem in my project in the firebase_core

### DISCUSSION THREAD

**Discussion by EngAhmedAsqein** [firebase/flutterfire#12362] [Q&A]
ERROR:C:\flutterPojects\firebase2\build\firebase_core\intermediates\runtime_library_classes_jar\debug\classes.jar: D8: java.lang.NullPointerException: Cannot invoke "String.length()" because "<parameter1>" is null

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeLibDexDebug'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Failed to transform classes.jar (project :firebase_core) to match attributes {artifactType=android-dex, asm-transformed-variant=NONE, com.android.build.api.attributes.AgpVersionAttr=7.3.0, com.android.build.api.attributes.BuildTypeAttr=debug, com.android.build.gradle.internal.attributes.VariantAttr=debug, dexing-enable-desugaring=true, dexing-enable-jacoco-instrumentation=false, dexing-is-debuggable=true, dexing-min-sdk=21, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for DexingWithClasspathTransform: C:\flutterPojects\firebase2\build\firebase_core\intermediates\runtime_library_classes_jar\debug\classes.jar.
         > Error while dexing.

**Comment by iamansuman** [comment]
Use JDK 19 (or anything below 20)
Configure your Gradle version by looking up [Gradle Compatibility Matrix](https://docs.gradle.org/current/userguide/compatibility.html#java)

> In case you're using JDK 19, go for Gradle 7.6

Go to %USERPROFILE% (Which is usually C:/Users/\<your windows username\>)
Delete the .gradle folder (you can also delete the .android folder)

Now go to `your project root folder > android > gradle > wrapper > gradle-wrapper.properties`

Change the distributionUrl as `https\://services.gradle.org/distributions/gradle-7.6-all.zip` (Check your Gradle version first and replace 7.6 with your suitable version, if you're not using JDK 19)
